test/date_parameter_test.py

```diff
@@ -0,0 +1,140 @@
+
+class DateTask(luigi.Task):
+    day = luigi.DateParameter()
+
+
+class DateHourTask(luigi.Task):
+    dh = luigi.DateHourParameter()
+
+
+class DateMinuteTask(luigi.Task):
+    dm = luigi.DateMinuteParameter()
+
+
+class MonthTask(luigi.Task):
+    month = luigi.MonthParameter()
+
+
+class YearTask(luigi.Task):
+    year = luigi.YearParameter()
+
+
+class DateTest(unittest.TestCase):
+    def test_parse(self):
+        d = luigi.DateParameter().parse('2015-04-03')
+        self.assertEqual(d, datetime.date(2015, 4, 3))
+
+    def test_serialize(self):
+        d = luigi.DateParameter().serialize(datetime.date(2015, 4, 3))
+        self.assertEqual(d, '2015-04-03')
+
+    def test_parse_interface(self):
+        task = luigi.interface.ArgParseInterface().parse(["DateTask", "--day", "2015-04-03"])[0]
+        self.assertEqual(task.day, datetime.date(2015, 4, 3))
+
+    def test_serialize_task(self):
+        t = DateTask(datetime.date(2015, 4, 3))
+        self.assertEqual(str(t), 'DateTask(day=2015-04-03)')
+
+
+class DateHourTest(unittest.TestCase):
+    def test_parse(self):
+        dh = luigi.DateHourParameter().parse('2013-02-01T18')
+        self.assertEqual(dh, datetime.datetime(2013, 2, 1, 18, 0, 0))
+
+    def test_serialize(self):
+        dh = luigi.DateHourParameter().serialize(datetime.datetime(2013, 2, 1, 18, 0, 0))
+        self.assertEqual(dh, '2013-02-01T18')
+
+    def test_parse_interface(self):
+        task = luigi.interface.ArgParseInterface().parse(["DateHourTask", "--dh", "2013-02-01T18"])[0]
+        self.assertEqual(task.dh, datetime.datetime(2013, 2, 1, 18, 0, 0))
+
+    def test_serialize_task(self):
+        t = DateHourTask(datetime.datetime(2013, 2, 1, 18, 0, 0))
+        self.assertEqual(str(t), 'DateHourTask(dh=2013-02-01T18)')
+
+
+class DateMinuteTest(unittest.TestCase):
+    def test_parse(self):
+        dm = luigi.DateMinuteParameter().parse('2013-02-01T18H42')
+        self.assertEqual(dm, datetime.datetime(2013, 2, 1, 18, 42, 0))
+
+    def test_parse_padding_zero(self):
+        dm = luigi.DateMinuteParameter().parse('2013-02-01T18H07')
+        self.assertEqual(dm, datetime.datetime(2013, 2, 1, 18, 7, 0))
+
+    def test_serialize(self):
+        dm = luigi.DateMinuteParameter().serialize(datetime.datetime(2013, 2, 1, 18, 42, 0))
+        self.assertEqual(dm, '2013-02-01T18H42')
+
+    def test_serialize_padding_zero(self):
+        dm = luigi.DateMinuteParameter().serialize(datetime.datetime(2013, 2, 1, 18, 7, 0))
+        self.assertEqual(dm, '2013-02-01T18H07')
+
+    def test_parse_interface(self):
+        task = luigi.interface.ArgParseInterface().parse(["DateMinuteTask", "--dm", "2013-02-01T18H42"])[0]
+        self.assertEqual(task.dm, datetime.datetime(2013, 2, 1, 18, 42, 0))
+
+    def test_serialize_task(self):
+        t = DateMinuteTask(datetime.datetime(2013, 2, 1, 18, 42, 0))
+        self.assertEqual(str(t), 'DateMinuteTask(dm=2013-02-01T18H42)')
+
+
+class MonthTest(unittest.TestCase):
```

Comment:
Either stick with the "DateBlahTest" convention or change the other examples. :)